### PR TITLE
feat(document): add support for pathsToSkip in validate and validateSync

### LINKED ISSUE #10353: [Feature] Document#validate(...)  `pathsToSkip` option to exclude fields
Currently, we can make an inclusive list of fields that we want to validate using `pathsToValidate`.

It would be nice if we could have an option to skip specific fields `pathsToSkip`, and a user can either select fields to skip, or fields to validate, never both.

@vkarpov15 What do you think? I'd happily work on that if accepted.

### PR DESCRIPTION
**Pull request by AbdelrahmanHafez**
fixes #10353

This still doesn't add support for `pathsToSkip` for Model.validate(...)` because the current parameters are:
```js
Model.validate = function validate(obj, pathsToValidate, context, callback) { ... }
```
We can not add it as a property to any of those parameters, so I'll defer implementing it until #10367 is implemented, then it'll make our lives much easier to extend our API.

### CODE REVIEW

File: lib/document.js [hunk 1]
```diff
@@ -2250,6 +2251,9 @@ Document.prototype.validate = function(pathsToValidate, options, callback) {
     options = pathsToValidate;
     pathsToValidate = null;
   }
+  if (options && typeof options.pathsToSkip === 'string') {
+    options.pathsToSkip = options.pathsToSkip.split(' ');
```
Comment (vkarpov15) [inline lib/document.js:2255]: For perf reasons, let's make this `options.pathsToSkip = options.pathsToSkip.indexOf(' ') === -1 ? [options.pathsToSkip] : options.pathsToSkip.split(' ');` . `split()` is very slow unfortunately, even if there's nothing to split.

File: lib/document.js [hunk 2]
```diff
@@ -2646,8 +2649,12 @@ Document.prototype.validateSync = function(pathsToValidate, options) {
     shouldValidateModifiedOnly = this.$__schema.options.validateModifiedOnly;
   }
 
+  let pathsToSkip = get(options, 'pathsToSkip');
+
   if (typeof pathsToValidate === 'string') {
     pathsToValidate = pathsToValidate.split(' ');
+  } else if (typeof pathsToSkip === 'string') {
+    pathsToSkip = pathsToSkip.split(' ');
```
Comment (vkarpov15) [inline lib/document.js:2657]: Same for this one.

File: lib/document.js [hunk 3]
```diff
@@ -2646,8 +2649,12 @@ Document.prototype.validateSync = function(pathsToValidate, options) {
     shouldValidateModifiedOnly = this.$__schema.options.validateModifiedOnly;
   }
 
+  let pathsToSkip = get(options, 'pathsToSkip');
```
Comment (vkarpov15) [inline lib/document.js:2652]: Let's just do `pathsToSkip = options.pathsToSkip`, `get()` is handy for dotted paths and cases where we don't know whether `options` exists, but it also has some non-trivial performance overhead.

**Review by IslandRhythms — COMMENTED** [review]
ditto everything Val said

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:

**Comment by AbdelrahmanHafez** [comment]
Happy to help 👍

**Pull request merged**